lib/main.ts

```diff
@@ -497,19 +499,30 @@ export class Transpiler {
     this.emit(translated);
   }
 
-  static DART_KEYWORDS =
-      ('abstract as assert async await break case catch class const continue ' +
-       'default deferred do dynamic else enum export extends external factory false final ' +
-       'finally for get if implements import in is library new null operator part rethrow return' +
-       ' set static super switch sync this throw true try typedef var void while with yield')
-          .split(/ /);
+  // Dart keyword list copied from
+  // https://www.dartlang.org/docs/dart-up-and-running/ch02.html#keywords
+  // Note the different kinds of keywords defined by the enum; key values
+  // match the footnote numbers given in the docs, or 0 if none given.
+  static DART_KEYWORDS : { [key:string] : DART_KEYWORD_KIND } = {
```

Comment:
Do we actually need anything beyond the RESERVED keywords? If not, I'd just drop the BUILTIN and LIMITED stuff.